underscore.comparison.islike.js

```diff
@@ -0,0 +1,70 @@
+/*
+* Tests if an object is like another. This means objects should follow the same
+* structure and arrays should contain the same types.
+*
+* E.g.
+*
+* _.islike(
+*   {name: "James", age: 10, hobbies: ["football", "computer games", "baking"]},
+*   {name: "", age: 0, hobbies: [""]}
+* )
+*/
+(function() {
+  // Establish the root object, `window` in the browser, or `require` it on the server.
+  if (typeof exports === 'object') {
+    _ = module.exports = require('underscore');
+  }
+
+    var islike = function(obj, pattern) {
+        if (typeof pattern === "function") {
+            return obj instanceof pattern;
+        }
+
+        if (typeof obj !== typeof pattern) return false;
+            if (pattern instanceof Array && !(obj instanceof Array)) return false;
```

Comment:
`_.isArray` is a good call. I don't think comparing constructors works because I'm only trying to reject arrays at this point.